src/compiler/checker.ts

```diff
@@ -7996,13 +8073,17 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return !(getObjectFlags(type) & ObjectFlags.Reference) || !isTypeReferenceNode(existing) || length(existing.typeArguments) >= getMinTypeArgumentCount((type as TypeReference).target.typeParameters);
         }
 
+        function getEnclosingDeclarationIgnoringFakeScope(enclosingDeclaration: Node) {
+            return findAncestor(enclosingDeclaration, n => !getNodeLinks(n).fakeScopeForSignatureDeclaration);

```

Comment:
Likewise, I think this should only need to check the immediate `enclosingDeclaration` now.

Reply:
Could be, though if that's the case, it'll only walk one node upward so I'm not sure I want to write it less conveniently...

Reply:
Switched this to a simple ternary.